src/node_contextify.cc

```diff
@@ -270,6 +271,83 @@ void ContextifyContext::IsContext(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void ContextifyContext::GetWrappedFunction(
+    const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Local<Context> context = env->context();
+  Isolate* isolate = env->isolate();
+
+  CHECK(args[0]->IsString());
+  Local<String> code = args[0].As<String>();
+
+  CHECK(args[1]->IsBoolean());
+  bool produce_cached_data = args[1]->IsTrue();
+
+  Local<Uint8Array> cached_data_buf;
+  if (!args[2]->IsUndefined()) {
+    CHECK(args[2]->IsUint8Array());
+    cached_data_buf = args[2].As<Uint8Array>();
+  }
+
+  // TODO(ryzokuken): Add support for sandboxing.
+
+  ScriptCompiler::CachedData* cached_data = nullptr;
+  if (!cached_data_buf.IsEmpty()) {
+    ArrayBuffer::Contents contents = cached_data_buf->Buffer()->GetContents();
+    uint8_t* data = static_cast<uint8_t*>(contents.Data());
+    cached_data = new ScriptCompiler::CachedData(
+      data + cached_data_buf->ByteOffset(), cached_data_buf->ByteLength());
```

Comment:
All this can really be done inside the if-block where you checked `args[2]`. That way you can also define `cached_data_buf` inside the block.

Reply:
I did it in this particular fashion to mimic what `New` did, so I have no preferences here whatsoever. Will do this.

Reply:
I see. In that case it's alright then.